Source/Core/DolphinQt/Debugger/MemoryWidget.cpp

```diff
@@ -689,6 +693,53 @@ void MemoryWidget::OnSetValue()
   Update();
 }
 
+void MemoryWidget::OnSetValueFromFile()
+{
+  if (!Core::IsRunning())
+    return;
+
+  auto target_addr = GetTargetAddress();
+
+  if (!target_addr.is_good_address)
+  {
+    ModalMessageBox::critical(this, tr("Error"), tr("Bad address provided."));
+    return;
+  }
+
+  if (!target_addr.is_good_offset)
+  {
+    ModalMessageBox::critical(this, tr("Error"), tr("Bad offset provided."));
+    return;
+  }
+
+  QString path = QFileDialog::getOpenFileName(this, tr("Select a file"), QDir::currentPath(),
+                                              tr("All files (*.*)"));
+  if (path.isNull())
+  {
+    ModalMessageBox::critical(this, tr("Error"), tr("No file provided."));
+    return;
+  }
```

Comment:
This triggers when you hit Cancel in the file dialog. It'd be better if that just silently returned, without an error message.